app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -18,16 +18,8 @@ def add_smart_proxy_filters(actions, options = {})
       self.smart_proxy_filter_actions ||= []
       self.smart_proxy_filter_actions.push(*actions)
 
-      prepend SmartProxyRequireSsl
-    end
-  end
-
-  module SmartProxyRequireSsl
-    def require_ssl?
-      if [self.smart_proxy_filter_actions].flatten.map(&:to_s).include?(action_name)
-        false
-      else
-        super
+      actions.each do |action|
+        Foreman::ForceSsl.add_allowed_http_path_regex!(/#{path_for(controller: controller_name, action)}/)
```

Comment:
This part confuses me. The Smart Proxy auth (generally) **must** happen over HTTPS and `smart_proxy_filter_actions` is really about excluding that for certain actions. I really wonder why we have this.

I would say that if HTTPS is present at all, Smart Proxy Auth should always happen over HTTPS.

I'm wondering if we can get rid of this somehow.

If not, Shouldn't it really escape things?
```suggestion
        Foreman::ForceSsl.add_allowed_http_path_regex!(Regexp.escape(path_for(controller: controller_name, action)))
```

Also, I think it would be better to do the `Regexp::union` here. Calling it multiple times results in a more complex regexp:
```
Regexp::union(Regexp::union(Regexp::union('a', 'b'), 'c'), 'd')
=> /(?-mix:(?-mix:a|b)|c)|d/
```

Reply:
Well if I read correctly it is for getting the message `"SSL is required - request from #{request.remote_ip}"` into the logs.
Which is probably not the best way to do it (but smart_proxy will not follow the redirect to https, so you won't even notice it did contacted foreman without the message, so it's somewhat usefull message)

Reply:
You partially correct. The templates module logs an error:
https://github.com/theforeman/smart-proxy/blob/7b3508a8444c04148f6c99f426a5de5d805f99a6/modules/templates/proxy_request.rb#L51

The registration module doesn't log an error but instead sends an error to the client:
https://github.com/theforeman/smart-proxy/blob/7b3508a8444c04148f6c99f426a5de5d805f99a6/modules/registration/registration_api.rb#L22-L30

Though I do think HTTP 500 codes may be logged.

So I would argue that sending a redirect is good enough and we don't need the additional complexity. Perhaps that should be a change we merge before merging this with a separate issue for changelog purposes?

Reply:
This doesn't concern templates I believe. But sure, having the redirect is good enough IMHO.
It might be good to log on the proxy side, but it should not be concern of Foreman (definitelly not for a huge price) - #9020

Reply:
Ok, so you may be right that it's about the ENC but there's https://github.com/theforeman/puppet-puppetserver_foreman/blob/616570d31f787b08640d2ffc3ee7c2d6850afd71/files/enc.rb#L268-L279 as well. Perhaps it's only the report processor itself since https://github.com/theforeman/puppet-puppetserver_foreman/blob/master/files/report.rb doesn't appear to check the response code.